src/Loader/AbstractFileLoader.php

```
@@ -57,20 +60,19 @@ protected function findRelativePath(string $path)
      */
     protected function findAbsolutePath(string $path)
     {
-        if (is_file($path) || is_dir($path)) {
-            return realpath($path);
+        if (file_exists($path)) {
+            return Filesystem::getRealPath($path);
```

I think for guaranteed BC, this also still needs to catch the exception and return false. 

This is because this method is used from `->supports()` (which just skips over any resource where this method returns `false`, so the later code that calls it and throws if it returns false is never called.).

To be honest I'm slightly struggling to think of examples where this would matter, and if starting from scratch I might be stricter than the old behaviour. But I do think there could be edge cases, maybe if a user has symlinks with strange permissions, that we currently ignore but would now throw on.

I still think [as my last comment](https://github.com/Behat/Gherkin/pull/367#discussion_r2127532007) that since this function explicitly follows the `realpath()` interface and the `false` return value is actually handled everywhere we call it, it is probably simpler to just keep using `realpath` directly rather than call our method just to catch the exception and return false again?

---

> I still think [as my last comment](https://github.com/Behat/Gherkin/pull/367#discussion_r2127532007) that since this function explicitly follows the `realpath()` interface and the `false` return value is actually handled everywhere we call it, it is probably simpler to just keep using `realpath` directly rather than call our method just to catch the exception and return false again?

totally agree here

---

Hmm, good point...

IMO, we should not suppress it, so that we are not modifying the existing behaviour (which might mean users see the warning, or get a failure from a custom error handler depending how they are running gherkin). Because I think it probably *is* unexpected that there's a file/directory inside your configured path(s) that exists but cannot be resolved to an absolute path. I just want to be cautious about breaking BC if someone is relying on that.

In a future major we could absolutely refactor this method to be stricter and to bubble the exception (in both cases) rather than returning `false`.